lib/topological_inventory/schema/default.rb

```diff
@@ -123,6 +124,106 @@ def add_tagging_collection(model, manager_ref: [:source_ref])
         end
       end
 
+      def add_service_instance_tasks
+        add_collection(:service_instance_tasks, inventory_collection_builder, {}, {:without_model_class => true}) do |builder|
+          add_default_properties(builder)
+          add_default_values(builder)
+
+          builder.add_dependency_attributes(
+            :service_instances => ->(persister) { [persister.collections[:service_instances]] }
+          )
+
+          save_block = lambda do |source, tasks_collection|
+            service_instance_tasks_custom_save(source, tasks_collection)
+          end
+
+          builder.add_properties(:custom_save_block => save_block)
+        end
+      end
+
+      def service_instance_tasks_custom_save(source, tasks_collection)
+        service_instance_collection = tasks_collection.dependency_attributes[:service_instances]&.first
+        src_refs = service_instance_collection&.data.to_a.collect { |inventory_object| inventory_object.source_ref }
+        return if src_refs.blank?
+
+        # Updating Tasks
+        # service_instance_tasks_update_effective()
+        service_instance_tasks_update_ineffective(source, src_refs)
+      end
+
+      def task_update_values(svc_instance_id, external_url, status, finished_timestamp)
+        {
+          :state  => finished_timestamp.blank? ? 'running' : 'completed',
+          :status => %w[error failed].include?(status) ? 'error' : 'ok', # TODO: ansible-tower specific, normalize in collector
+          :context => {
+            :remote_status => status,
+            :service_instance => {
+              :id => svc_instance_id,
+              :url => external_url
+            }
+          }
+        }
+      end
+
+      # This method is updating one by one using ActiveRecord
+      def service_instance_tasks_update_ineffective(source, svc_instances_source_ref)
+        service_instances = ServiceInstance.where(:source_id => source.id, :source_ref => svc_instances_source_ref)
```

Comment:
Since this is a dependent collection on service_instances we should have the ids here from the inventory collection but we can look at that in a followup